### ISSUE TITLE
NC15: General error: 1364 Field 'id' doesn't have a default value

### ISSUE THREAD

**Issue by CaptInsano** [ChristophWurst/twofactor_admin#26]
I just updated to NC15 and got locked out of my Nextcloud instance without access to my backup codes:

After reading everything I could: I found this app, I tried to install via occ and got the same error as listed in another issue: "1067 Invalid default value for 'expires'". I followed your advice in that issue and issued the 

`ALTER TABLE oc_twofactor_admin_codes CHANGE COLUMN `expires` expires BIGINT UNSIGNED NOT NULL`

command to my mariadb docker to allow me enable the app successfully however now I have a new issue:

When I issue the command "./occ twofactor:admin:generate-code USERNAME" I get the following error:

```
In AbstractMySQLDriver.php line 115:
  An exception occurred while executing 'INSERT INTO `oc_twofactor_admin_codes` (`user_id`, `code`, `expires`) VALUES(?, ?, ?)' with par
  ams ["USERNAME", "015463", 1544656964]:
  SQLSTATE[HY000]: General error: 1364 Field 'id' doesn't have a default value
In PDOStatement.php line 107:
  SQLSTATE[HY000]: General error: 1364 Field 'id' doesn't have a default value
In PDOStatement.php line 105:
  SQLSTATE[HY000]: General error: 1364 Field 'id' doesn't have a default value
```

Any help would be greatly appreciated as I am now locked out of my server

**Comment by ChristophWurst** [comment]
Dup of #20 or #24

**Comment by ChristophWurst** [comment]
Is this mariadb 10.2 or 10.3?

**Comment by CaptInsano** [comment]
Seeing as the error was different, I thought you would want a new issue report.

Server:                 MariaDB
Server version:         10.3.11-MariaDB-1:10.3.11+maria~bionic-log mariadb.org binary distribution

**Comment by ChristophWurst** [comment]
I generally do, but in this case it is unfortunately a general incompatibility with new versions of mariadb.

> Any help would be greatly appreciated as I am now locked out of my server

What 2FA providers do you use, btw? I'm currently preparing releases for totp and u2f that add the ability to disable those providers for a single user via the CLI as well. So this might be of interest for you.

https://github.com/nextcloud/twofactor_totp/pull/360 <- testing welcome in case you have it installed

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/twofactor_u2f/pull/311 <- testin welcome if that's what you're using (too)

**Comment by CaptInsano** [comment]
I fixed my issue by using HeidiSQL viewer (my mysql-foo is not strong!) and setting the "id" value in the oc_twofactor_admin_codes table to "incremental increase". Then the occ twofactot:admin:generate-code command worked.

I have been using the TOTP 2FA provider for well over a year successfully.

When I updated from NC 14.1 to NC 15 via the web interface it was disabled for some reason but I was not alerted of same and that was how I became locked out. I tried enabling it again via occ but I was advised that it was not compatible with NC15. That was when I went scouring the internet for solutions.

Anyway, I am back in my server but certainly not the seemless update that we are used to!
